src/api/java/com/github/minecraftschurlimods/arsmagicalegacy/api/data/ObeliskFuelProvider.java

```diff
@@ -0,0 +1,153 @@
+package com.github.minecraftschurlimods.arsmagicalegacy.api.data;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+import net.minecraft.data.DataGenerator;
+import net.minecraft.data.DataProvider;
+import net.minecraft.data.HashCache;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.tags.Tag;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.crafting.Ingredient;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public abstract class ObeliskFuelProvider implements DataProvider {
+    private static final Gson GSON = new GsonBuilder().setPrettyPrinting().create();
+    private static final Logger LOGGER = LogManager.getLogger();
+    protected final DataGenerator generator;
+    private final String namespace;
+    private final Map<ResourceLocation, JsonObject> data = new HashMap<>();
+
+    public ObeliskFuelProvider(String namespace, DataGenerator generator) {
+        this.namespace = namespace;
+        this.generator = generator;
+    }
+
+    protected abstract void createFuels();
+
+    @Override
+    public void run(HashCache pCache) {
+        createFuels();
+        data.forEach((resourceLocation, jsonObject) -> save(pCache, jsonObject, generator.getOutputFolder().resolve("data/" + resourceLocation.getNamespace() + "/obelisk_fuel/" + resourceLocation.getPath() + ".json")));
+    }
+
+    @Override
+    public String getName() {
+        return "Obelisk Fuels";
```

Comment:
The name should be different for each instance with a different output namespace